[Dense Ores 1.6.1] Crash on loading

I crash on loading with the following crash log: http://paste.ee/p/AsQPd

The relevant lines seem to be:

Caused by: java.lang.NoClassDefFoundError: ic2/api/recipe/IRecipeInput
	at com.rwtema.denseores.modintegration.ModIntegration.<clinit>(ModIntegration.java:19)
	at com.rwtema.denseores.DenseOresMod.init(DenseOresMod.java:37)


I do not have IC2 installed.

--- a/src/main/java/com/rwtema/denseores/modintegration/IC2Compat.java
+++ b/src/main/java/com/rwtema/denseores/modintegration/IC2Compat.java
@@ -2,6 +2,8 @@ package com.rwtema.denseores.modintegration;
 
 import com.rwtema.denseores.DenseOre;
 import cpw.mods.fml.common.Loader;
+import cpw.mods.fml.common.Optional;
+import cpw.mods.fml.common.Optional.Method;
 import ic2.api.recipe.RecipeInputItemStack;
 import ic2.api.recipe.Recipes;
 import net.minecraft.item.ItemStack;
@@ -17,7 +19,8 @@ public class IC2Compat implements ModInterface {
             registerOre_do(ore, denseOre, originalOre);
     }
 
-    private static void registerOre_do(DenseOre ore, ItemStack denseOre, ItemStack originalOre) {
+    @Method(modid = "IC2")
+    private void registerOre_do(DenseOre ore, ItemStack denseOre, ItemStack originalOre) {
         if(Recipes.scrapboxDrops == null || Recipes.macerator == null)
             return;
 

--- a/src/main/java/com/rwtema/denseores/modintegration/ThaumcraftCompat.java
+++ b/src/main/java/com/rwtema/denseores/modintegration/ThaumcraftCompat.java
@@ -2,6 +2,7 @@ package com.rwtema.denseores.modintegration;
 
 import com.rwtema.denseores.DenseOre;
 import cpw.mods.fml.common.Loader;
+import cpw.mods.fml.common.Optional;
 import net.minecraft.item.ItemStack;
 import thaumcraft.api.ThaumcraftApi;
 import thaumcraft.api.aspects.Aspect;
@@ -11,7 +12,8 @@ import java.util.Map;
 
 public class ThaumcraftCompat implements ModInterface {
 
-    private static void registerOre_do(DenseOre ore, ItemStack denseOre, ItemStack originalOre) {
+    @Method(modid = "Thaumcraft")
+    private void registerOre_do(DenseOre ore, ItemStack denseOre, ItemStack originalOre) {
         AspectList aspectList = new AspectList(originalOre);
         if (aspectList.visSize() == 0)
             return;

